### ISSUE TITLE
Slow Git can cause Exception in diff panel for submodules

### ISSUE THREAD

**Issue by gerhardol** [gitextensions/gitextensions#7335]
## Current behaviour
If Git is extremely slow (for instance due to the Symantec virus that slows down Git operations ten times or more), it is possible to see exceptions in the diff window instead of the diff.
This has occurred a couple of times for me, I have not tried to provoke it though.

## Expected behaviour
Show the diff of the selected file

## Steps to reproduce

I have tried to insert delays in a proper setup without Symantec but not recreated the problem.
The submodule will likely have to be large too.

Working directory changes, in order:
 * A submodule with an incorrect commit checked out (as well as a few changed files so it is dirty)
 * A changed file

Select Worktree artificial commit, then select the file before the submodule status is displayed.

## Did this work in previous version of GitExtensions

Probably not

## Environment
- Git Extensions 3.3.0
- Build 395cf7af41c8bdbfe3d4ef421608623e23acf1e7
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4018.0
- DPI 96dpi (no scaling)

## Diagnostics

```
System.ArgumentNullException: Value cannot be null.

Parameter name: status

   at ResourceManager.LocalizationHelpers.ProcessSubmoduleStatus(GitModule module, GitSubmoduleStatus status) in F:\dev\gc\gitextensions_4\ResourceManager\LocalizationHelpers.cs:line 134
   at GitUI.GitUIExtensions.GetSelectedPatch(FileViewer diffViewer, ObjectId firstRevision, ObjectId secondRevision, GitItemStatus file) in F:\dev\gc\gitextensions_4\GitUI\GitUIExtensions.cs:line 56
   at GitUI.GitUIExtensions.<>c__DisplayClass3_0.<ViewChangesAsync>b__0() in F:\dev\gc\gitextensions_4\GitUI\GitUIExtensions.cs:line 121
   at GitCommands.AsyncLoader.<>c__DisplayClass11_0`1.<LoadAsync>b__0(CancellationToken token) in F:\dev\gc\gitextensions_4\GitCommands\AsyncLoader.cs:line 44
   at GitCommands.AsyncLoader.<LoadAsync>d__12`1.MoveNext() in F:\dev\gc\gitextensions_4\GitCommands\AsyncLoader.cs:line 83
```

**Comment by gerhardol** [comment]
The Task is likely cancelled, but I do not see how

```
        private static string GetSelectedPatch(
            [NotNull] this FileViewer diffViewer,
            [CanBeNull] ObjectId firstRevision,
            [CanBeNull] ObjectId secondRevision,
            [NotNull] GitItemStatus file)
        {
            if (!file.IsTracked)
            {
                var fullPath = Path.Combine(diffViewer.Module.WorkingDir, file.Name);
                if (Directory.Exists(fullPath) && GitModule.IsValidGitWorkingDir(fullPath))
                {
                    // git-status does not detect details for untracked and git-diff --no-index will not give info
                    return LocalizationHelpers.GetSubmoduleText(diffViewer.Module, file.Name.TrimEnd('/'), "");
                }
            }

            if (file.IsSubmodule && file.GetSubmoduleStatusAsync() != null)
            {
                 return LocalizationHelpers.ProcessSubmoduleStatus(diffViewer.Module, ThreadHelper.JoinableTaskFactory.Run(() => file.GetSubmoduleStatusAsync()));
            }
```

**Comment by RussKie** [comment]
@sharwell any thoughts?

**Comment by gerhardol** [comment]
Fixed in #7926
